libretro-buildbot-recipe.sh

```diff
@@ -750,9 +750,9 @@ buildbot_pull(){
 					cd $DIR
 					git checkout "$BRANCH"
 				elif [ ! -z "$GIT_BRANCH" ]; then
-					git clone -b "$GIT_BRANCH" "$URL" "$DIR" --depth=1
+					git clone -b "$GIT_BRANCH" "$URL" "$DIR" --single-branch
 				else
-					git clone -b master "$URL" "$DIR" --depth=1
+					git clone -b master "$URL" "$DIR" --single-branch
```

Comment:
The `--single-branch` change seems like a viable change. It is outside the scope of updating this to use the upstream repo, however. I'll let others decide whether we should keep the diff as is or recommend splitting it into two different PRs.

Reply:
Thanks, let me know :)

Reply:
This will break the script....

Its already cloning a single branch, not every repo is using `master` and it will be slower without`--depth=1`.